server/game/cards/07-GR/DarkMemento.js

```diff
@@ -0,0 +1,65 @@
+const Card = require('../../Card.js');
+
+class DarkMemento extends Card {
+    // At the start of your turn, if you are not haunted, discard
+    // cards from the top of your deck until you are haunted.
+    //
+    // Scrap: Discard the top card from a player‘s deck. If that player is
+    // not haunted, repeat this effect.
+    setupCardAbilities(ability) {
+        this.reaction({
+            when: {
+                onBeginRound: (_, context) =>
+                    context.player === this.game.activePlayer && !context.player.isHaunted()
+            },
+            condition: (context) => context.player.deck.length > 0,
+            gameAction: ability.actions.discard((context) => ({
+                target: context.player.deck[0]
+            })),
+            then: (preThenContext) => ({
```

Comment:
Do we need to use this recursion? 
Can't we discard as much as needed, like:
```
 gameAction: ability.actions.discard((context) => ({
    target: context.player.deck.slice(0, 0, 10 - content.player.discard.length)
}))
```            
wdyt?

Reply:
I was worried there could be interruptions on each individual discard that could end up canceling the rest, or moving the discarded card to a different zone (like Bottomless Depths does for cards discarded from your hand).  If you're not worried about that happening in future cards we can simplify.

Reply:
Also, the "10" doesn't work because of that other card that changes how haunted is calculated.

Reply:
> Also, the "10" doesn't work because of that other card that changes how haunted is calculated.

ouch ouch ouch :( So true.